src/Dropdown/Dropdown.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * Copyright 2017 CA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import React, { Component } from 'react';
+import { findDOMNode } from 'react-dom';
+import { generateId } from '../utils';
+import Root, { PopoverTrigger as DropdownTrigger } from '../Popover';
+import DropdownContent from './DropdownContent';
+
+type Item = {
+  iconEnd?: React$Element<*>,
+  iconStart?: React$Element<*>,
+  disabled?: boolean,
+  divider?: boolean,
+  onClick?: (event: Object) => void,
+  render?: (item: Object, props: Object, theme: Object) => React$Element<*>,
+  secondaryText?: MnrlReactNode,
+  text?: MnrlReactNode,
+  variant?: 'regular' | 'danger' | 'success' | 'warning'
+};
+
+type Props = {
+  /** Trigger for the Dropdown */
+  children: MnrlReactNode,
+  /** Open the Dropdown immediately upon initialization */
+  defaultIsOpen?: boolean,
+  /** Disable the Dropdown */
+  disabled?: boolean,
+  /** Data from which the [Menu](./menu#data) will be constructed */
+  data: Array<Object>,
+  /** For use with controlled components, in which the app manages Dropdown state */
+  isOpen?: boolean,
+  /** Plugins used to alter behavior. See [PopperJS docs](https://popper.js.org/popper-documentation.html#modifiers) for options. */
+  modifiers?: Object,
+  /** Called when Dropdown is closed */
+  onClose?: (event: Object) => void,
+  /** Called when Dropdown is opened */
+  onOpen?: (event: Object) => void,
+  /** Placement of the Dropdown menu */
+  placement?:
+    | 'bottom-end'
+    | 'bottom-start'
+    | 'left-end'
+    | 'left-start'
+    | 'right-end'
+    | 'right-start'
+    | 'top-end'
+    | 'top-start',
+  /** Focus trigger after selecting an item */
+  restoreFocus?: boolean,
+  /** Display a wider Dropdown menu */
```

Comment:
Does this mean it's really wide, like full-width wide? Do we need to specify how much wider?